File: examples/sync/mexc_public.py
import dcex


def main() -> None:
    client = dcex.mexc()

    exchange_info = client.get_spot_exchange_info(product_symbol="BTC-USDT-SPOT")
    print(exchange_info)

    orderbook = client.get_contract_depth(product_symbol="BTC-USDT-SWAP", limit=5)
    print(orderbook)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/mexc_public.rs
use std::time::Duration;

use dcex::exchanges::mexc::MexcClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = MexcClient::public(Duration::from_secs(10))?;
    let response = client.get_spot_orderbook("BTC-USDT-SPOT").limit(5).await?;
    println!("{}", response.data);
    Ok(())
}
